linera-execution/src/transaction_tracker.rs

```diff
@@ -21,18 +22,36 @@ pub struct TransactionTracker {
     oracle_responses: Vec<OracleResponse>,
     outcomes: Vec<ExecutionOutcome>,
     next_message_index: u32,
+    pending_application_blobs: Arc<BTreeMap<BlobId, Blob>>,
```

Comment:
If these are all application descriptions, why not make it type safe: `BTreeMap<ApplicatoinId, ApplicationDescription>`?

But actually: Why do we need this here for application descriptions but not e.g. for data blobs published in this block?